src/main/java/com/minecolonies/coremod/datalistener/QuestJsonListener.java

```diff
@@ -0,0 +1,366 @@
+            if (!effectRegistry.containsKey(effectID))
+            {
+                throw new Exception("Unkown/unregistered quest effect: for " + effectID);
+            }
+
+            effectData.put(effectID, effectJsonObj);
+        }*/
+    }
+
+    // Unused yet
+    private static Function<IColony, List<ITriggerReturnData>> parseTriggerOrder(final ResourceLocation questId, final String order, final List<IQuestTrigger> triggers)
+    {
+        // Default and.
+        if (order.isEmpty())
+        {
+            return colony -> {
+                final List<ITriggerReturnData> returnList = new ArrayList<>();
+
+                for (final IQuestTrigger trigger: triggers)
+                {
+                    ITriggerReturnData returnData = trigger.isFulfilledForColony(colony);
+                    if (returnData.isPositive())
+                    {
+                        returnList.add(returnData);
+                    }
+                    else
+                    {
+                        return null;
+                    }
+                }
+                return returnList;
+            };
+        }
+
+        // Test
+        //order = "(2 && (1 || 3)) && ((1 && 2) || 3)";
+        // Replace whitespaces
+        //order = order.replaceAll("\\s+", "");
+
+        // Split by words and braces, but keep the chars
+        final List<String> values = Arrays.asList(order.replaceAll("\\s+","").split("((?<=\\w)|(?=\\w)|(?<=[)(])|(?=[)(]))"));
+
+        final List<String> types = new ArrayList<>();
+        for (String value : values)
+        {
+            try
+            {
+                // Try parsing to number. We only allow numbers.
+                Integer.parseInt(value);
+                if (!types.contains(value))
+                {
+                    types.add(value);
+                }
+            }
+            catch (final Exception ex)
+            {
+                //ignore
+            }
+        }
+
+        if (types.size() != triggers.size())
+        {
+            Log.getLogger().error("Failed to Parse Quest Triggers. Mismatch number of triggers in order for quest: " + questId.toString());
+            return null;
+        }
+
+        final Map<String, IQuestTrigger> triggerMap = new HashMap<>();
+        for (int i = 0; i < triggers.size(); i++)
+        {
+            triggerMap.put(String.valueOf(i+1), triggers.get(i));
+        }
+
+        return colony -> evaluate(colony, triggerMap, new ArrayList<>(values), new ArrayList<>());
+    }
+
+    /**
+     * Recursively parses a string condition to a predicate
+     *
+     * @param colony the colony.
+     * @param data split string data
+     * @return predicate from data
+     */
+    private static List<ITriggerReturnData> evaluate(final IColony colony, final Map<String, IQuestTrigger> triggerMap, final List<String> data, final List<ITriggerReturnData> lastReturnData)
```

Comment:
Wouldn't it be nicer to datapack authors to use existing conventions (e.g. advancement triggers or list-of-lists)?